app/scripts/modules/amazon/src/serverGroup/details/resize/AmazonResizeServerGroupModal.tsx

```diff
@@ -0,0 +1,424 @@
+  CheckboxInput,
+  TaskReason,
+  NgReact,
+  ICapacity,
+} from '@spinnaker/core';
+
+import { AwsModalFooter } from 'amazon/common';
+import { IAmazonServerGroup } from 'amazon/domain';
+
+export interface IAmazonResizeServerGroupModalProps extends IModalComponentProps {
+  application: Application;
+  serverGroup: IAmazonServerGroup;
+}
+
+export interface IAmazonResizeServerGroupModalState {
+  advancedMode: boolean;
+  interestingHealthProviderNames: string[]; // managed by a separate component
+  initialValues: IAmazonResizeServerGroupValues;
+  taskMonitor: TaskMonitor;
+  platformHealthOnlyShowOverride?: boolean;
+}
+
+export interface IAmazonResizeServerGroupValues {
+  min: number;
+  max: number;
+  desired: number;
+  enforceCapacityConstraints: boolean;
+  reason?: string;
+}
+
+export interface IResizeJob extends IServerGroupJob {
+  constraints?: { capacity: ICapacity };
+  reason?: string;
+  interestingHealthProviderNames: string[];
+}
+
+export class AmazonResizeServerGroupModal extends React.Component<
+  IAmazonResizeServerGroupModalProps,
+  IAmazonResizeServerGroupModalState
+> {
+  public static defaultProps: Partial<IAmazonResizeServerGroupModalProps> = {
+    closeModal: noop,
+    dismissModal: noop,
+  };
+
+  private formikRef = React.createRef<Formik<IAmazonResizeServerGroupValues>>();
+
+  public static show(props: IAmazonResizeServerGroupModalProps): Promise<IResizeJob> {
+    const modalProps = {};
+    return ReactModal.show(AmazonResizeServerGroupModal, props, modalProps);
+  }
+
+  constructor(props: IAmazonResizeServerGroupModalProps) {
+    super(props);
+
+    const { minSize, maxSize, desiredCapacity } = props.serverGroup.asg;
+    const { attributes } = props.application;
+    this.state = {
+      advancedMode: minSize !== maxSize,
+      initialValues: {
+        min: minSize,
+        max: maxSize,
+        desired: desiredCapacity,
+        enforceCapacityConstraints: false,
+      },
+      taskMonitor: new TaskMonitor({
+        application: props.application,
+        title: 'Resizing your server group',
+        modalInstance: TaskMonitor.modalInstanceEmulation(() => this.props.dismissModal()),
+        onTaskComplete: () => this.props.application.serverGroups.refresh(),
+      }),
+      platformHealthOnlyShowOverride: attributes.platformHealthOnlyShowOverride,
+      interestingHealthProviderNames:
+        attributes.platformHealthOnlyShowOverride && attributes.platformHealthOnly ? ['Amazon'] : null,
+    };
+  }
+
+  private validate = (
+    values: IAmazonResizeServerGroupValues,
+  ): Partial<FormikErrors<IAmazonResizeServerGroupValues>> => {
+    if (!this.state.advancedMode) {
+      return {};
+    }
+    const errors: Partial<FormikErrors<IAmazonResizeServerGroupValues>> = {};
+    const { min, max, desired } = values;
+    // try to only show one error message at a time
+    if (min > max && min > desired) {
+      errors.min = 'cannot be larger than Max/Desired';
```

Comment:
```suggestion
      errors.min = 'Min cannot be larger than Max/Desired';
```

Reply:
(discussed offline; horizontal space is really limited for display of this error label, so dropping leading text. Same applies to other suggestions)